include/elements/2d/quadrilateral_bspline_element.tcc

```diff
@@ -334,4 +430,49 @@ double mpm::QuadrilateralBSplineElement<Tdim, Tpolynomial>::gradient(
                            index + 1);
 
   return value;
+}
+
+//! Function to check if particle is lying on the region where kernel
+//! correction is necessary
+template <unsigned Tdim, unsigned Tpolynomial>
+bool mpm::QuadrilateralBSplineElement<
+    Tdim, Tpolynomial>::kernel_correction_region(const VectorDim& xi) const {
+  bool status = false;
+  // Exit if kernel_correction_ is false
+  if (!kernel_correction_) return status;
+
+  // Loop over dimension
+  for (unsigned i = 0; i < Tdim; ++i) {
+    // First loop over local nodes
+    for (unsigned n = 0; n < this->nfunctions_local(); ++n) {
+      // If it is located at the lower or uppermost boundary
+      if (this->node_type_[n][i] == 1 || this->node_type_[n][i] == 4) {
+        status = true;
+        goto breakout;
```

Comment:
![XnPXh](https://user-images.githubusercontent.com/24325772/191285429-25251ba9-9b58-4af3-ac5f-6017d06c0661.png)

Reply:
> ![XnPXh](https://user-images.githubusercontent.com/24325772/191285429-25251ba9-9b58-4af3-ac5f-6017d06c0661.png)

I could not resist the joke after seeing so many goto's statements jajajajaja

Reply:
Hahahha. This goto is a safe one. This is used to exit double `for` loop since I just need one `true` statement to identify the kernel correction.

Reply:
I know! This is one of these rare occasions where the best solution is a goto statement :)